.github/workflows/tidy-modern.yml

```diff
@@ -0,0 +1,59 @@
+# This is a basic workflow to help you get started with Actions
```

Comment:
@shrijitsingh99 Apologies for pinging you again. Thanks in advance

Reply:
No issues.
The current commit has an invalid syntax. 
The previous version has the correct syntax but had some errors.

Something on the lines of this should work (I haven't tested this out).

```yaml
name: tidy-modern

on:
  push:
  pull_request:

jobs:
  tidy:
    runs-on: ubuntu-latest
    
    steps:
    - uses: actions/checkout@v2

    - name: Run clang-tidy
      run: |
          ./.github/workflows/clang-tidy-modern.sh
          git diff > tidy.patch
          cat tidy.patch
          [ ! -s formatting.patch ]
 
```

Reply:
Looks clean and simple, I'll revise and give it a try.

Reply:
As I thought might happen:
```
./.github/workflows/clang-tidy-modern.sh: line 2: run-clang-tidy: command not found

```
@shrijitsingh99 I guess this will require setting up a containerized environment that install `clang-tidy`?

Reply:
No need for docker currently.
This should work:

```yaml

name​: ​tidy-modern​

​on​:
  ​push​:
  ​pull_request​:

​jobs​:
  ​tidy​:
    ​runs-on​: ​ubuntu-latest​
    
    ​steps​:
    - ​uses​: ​actions/checkout@v2​

    - ​name​: ​Run clang-tidy​
      ​run​: ​|​
          apt update
          apt install clang-tidy
​          ./.github/workflows/clang-tidy-modern.sh​
​          git diff > tidy.patch​
​          cat tidy.patch​
​          [ ! -s formatting.patch ]​
​
```

Reply:
@shrijitsingh99 @kunaltyagi Thanks, it finally worked. Here's the [log](https://github.com/gnawme/pcl/actions/runs/170646180).